### PR TITLE
IndentationTypeFixer - fix handling 2 spaces indent

### PR DESCRIPTION
**Pull request by kubawerlos**
For https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3837

### CODE REVIEW

File: src/Fixer/Whitespace/IndentationTypeFixer.php [hunk 1]
```diff
@@ -144,4 +144,19 @@ static function (array $matches) use ($indent) {
...
         return new Token([T_WHITESPACE, $newContent]);
     }
+
+    /**
+     * @param string $content
+     * @param string $indent
+     *
+     * @return string mixed
+     */
+    private function getExpectedIndent($content, $indent)
```
Comment (AdryanSignor) [inline src/Fixer/Whitespace/IndentationTypeFixer.php]: There isn't **$this** context within the function. Can you use static declaration?
Call getExpectedIndent with **self** token within anonymous function for Preg::replaceCallback, will
prevent anonymous function relocation.

Comment (keradus) [reply src/Fixer/Whitespace/IndentationTypeFixer.php]: static === global === evil
try to mock static method

no static here

File: tests/Fixer/Whitespace/IndentationTypeFixerTest.php [hunk 2]
```diff
@@ -324,4 +324,69 @@ static function ($key) {
...
+if (true) {
+  if (true) {
+\t(new stdClass())->foo(
+\t  'text',
+\t  'text2'
+\t);
+  }
+}",
+            ],
+            ['<?php
```
Comment (julienfalque) [inline tests/Fixer/Whitespace/IndentationTypeFixerTest.php]: This case is redundant with the subsequent one (as `doTest()` asserts that the expected part doesn't get changed by the fixer when used as input).

Comment (SpacePossum) [reply tests/Fixer/Whitespace/IndentationTypeFixerTest.php]: :+1:  force pushed it out

**Comment by SpacePossum** [comment]
Thank you @kubawerlos.

**Pull request merged**